bottom/protocol.py

```diff
@@ -1,29 +1,38 @@
 import asyncio
 import logging
 from bottom.unpack import unpack_command
+from typing import Optional, TYPE_CHECKING
+
+if TYPE_CHECKING:
```

Comment:
What other options are there instead of the circular import just to get a reference to `Client`?  I don't like circular imports[0] (module-only or otherwise) and I'll use `Any` before I add the headache of tracing imports when I refactor code.  A stubs module?  Not a huge fan but I could at least stomach that.

[0] I'm still disappointed that PEP 484 so casually recommends circular imports for something core to the type annotations problem space.